Source/Core/Core/IOS/ES/ES.h

```diff
@@ -55,35 +55,23 @@ class ES final : public Device
   ReturnCode Close(u32 fd) override;
   IPCCommandResult IOCtlV(const IOCtlVRequest& request) override;
 
-  struct OpenedContent
-  {
-    bool m_opened = false;
-    u64 m_title_id = 0;
-    IOS::ES::Content m_content;
-    u32 m_position = 0;
-    u32 m_uid = 0;
-  };
-
   struct TitleImportContext
   {
-    IOS::ES::TMDReader tmd;
-    u32 content_id = 0xFFFFFFFF;
-    std::vector<u8> content_buffer;
-  };
+    void DoState(PointerWrap& p);
 
-  // TODO: merge this with TitleImportContext. Also reuse the global content table.
-  struct TitleExportContext
-  {
-    struct ExportContent
+    bool valid = false;
+    IOS::ES::TMDReader tmd;
+    std::array<u8, 16> key{};
+    struct ContentContext
     {
-      OpenedContent content;
+      bool valid = false;
+      u32 id = 0;
       std::array<u8, 16> iv{};
+      std::array<u8, 20> sha1{};
```

Comment:
Good catch -- I intended to change the import code to decrypt/hash each chunk of content in ImportContentData, instead of ImportContentEnd, to avoid keeping possibly large buffers in memory, but decided against putting it in this PR as it'd be an unrelated change.